### PR TITLE
Magento to Zoho CRM - Products

### PR DESCRIPTION
**Pull request by charlotta-frends**
#168 

Updated to work with the refresh token! The products are created in Zoho with the bare minimum of attributes (since Zoho does not require much), but I'm open to comments about if more mapping should be added.

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
  - [x] If applicable, the process name follows the naming convention "{source/format} to {destination/format} - {object}"
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots
- [x] Template parameters checked
  - [x] Parameter follows the CamelCase naming convention, abbreviations included, e.g. ConnectionString, BaseUrl, ApiKey
  - [x] Parameter descriptions are good and end with dots
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output, e.g. tokens in the header.
- [x] Process end points like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
  - [x] Errors are pointed downwards from their respective source, e.g. an exclusive decision
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Template files are going to the correct folder
- [x] Markdown paragraphs are separated with two newlines, so there is a blank line inbetween two paragraphs
- [x] If some system is integrated using e.g. a HTTP task (meaning we do not use a native connector for the system), then the system should be mentioned in additional systems used in metadata.json
- [x] If the template is handling some sample JSON data - make sure that the sample data is included in the long description (just one-two elements is enough)
- [x] If the template is assuming that there is a database table ready for data insertion / updates - include the table structure in the long description as a `CREATE TABLE` scipt.

## Summary by CodeRabbit

- **New Features**
	- Introduced a template for seamless integration between Magento and Zoho CRM for efficient product management.
	- Added metadata configuration to define integration specifics, including name and version.
	- Implemented a process template to facilitate product retrieval, insertion, and updates between platforms with robust error handling.

- **Enhancements**
	- Improved token management to ensure uninterrupted operations and secure authentication across platforms.
	- Enhanced error handling to allow continued processing despite individual product upload failures.

### CODE REVIEW

**Review by FrendsSarlinS — CHANGES_REQUESTED** [review]
Looks good to me and checks all the boxes! I just wonder, if we should indeed have some extra information on the products. Wouldn't need much, I'd expect to see the product price and the SKU on the products, but as there doesn't seem to be any proper fields for SKU on Zoho CRM, I'd say the price would be sufficient.

File: Templates/E-Commerce/Magento/Magento to Zoho CRM - Products/long-description.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- The Magento user is eligible to obtain an access token from Magento.
+- The Zoho CRM client ID and client secret are accessible.
+- The domain for the Zoho instance (e.g. EU or USA) is identified.
+- A Zoho CRM refresh token with an adequate scope has been generated using the "Zoho CRM - Exchange grant token for refresh token" template. For this template, the token should be configured with the scopes `ZohoCRM.modules.products.READ` and `ZohoCRM.modules.products.CREATE`.
+
+# Implementation and Usage Notes
+
+Before retrieving the Magento product data, an access token used for authorizing the request is obtained from the Magento API. 
+
+Accessing the Zoho CRM API also requires an access token, which is obtained with a separate refresh token provided in the process variables. To generate this refresh token, you can use the corresponding "Zoho CRM - Exchange grant token for refresh token" template.
```
Comment (FrendsSarlinS) [inline Templates/E-Commerce/Magento/Magento to Zoho CRM - Products/long-description.md:18]: In this paragraph, should we add the information about the refresh tokens not expiring? With this wording, it gives me the vibes that I need to provide a new refresh token every time I use this template.

**Comment by charlotta-frends** [comment]
> I just wonder, if we should indeed have some extra information on the products. Wouldn't need much, I'd expect to see the product price and the SKU on the products, but as there doesn't seem to be any proper fields for SKU on Zoho CRM, I'd say the price would be sufficient.

Yeah, there's no matching column for SKU in Zoho (the Product_Code seems to be the closest thing). And with the price already included, I don't really see other essential columns that could be mapped from the fetched Magento data. The other Magento product templates have the price book stuff in them, but as Zoho doesn't require adding products in a price book, I was thinking maybe that can be left out in this one. 🤔

**Review by FrendsSarlinS — APPROVED** [review]
LGTM after the recent changes! The unit price was already in as well.

**Pull request merged**